packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/controllers/ProjectRestController.java

```diff
@@ -0,0 +1,120 @@
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * REST Controller for the Project Endpoint.
+ *
+ * @author arichard
+ */
+@RestController
+@RequestMapping("/api/rest/projects")
+public class ProjectRestController {
+
+    private final IProjectApplicationService projectApplicationService;
+
+    public ProjectRestController(IProjectApplicationService projectApplicationService) {
+        this.projectApplicationService = Objects.requireNonNull(projectApplicationService);
+    }
+
+    @GetMapping
+    public ResponseEntity<List<RestProject>> getProjects() {
+        var restProjects = this.projectApplicationService.findAll(PageRequest.of(0, 20))
+            .map(project -> new RestProject(project.id(), null, List.of(), null, null, project.name(), List.of()))
+            .toList();
+
+        return new ResponseEntity<>(restProjects, HttpStatus.OK);
+    }
+
+    @GetMapping(path = "/{projectId}")
+    public ResponseEntity<RestProject> getProjectById(@PathVariable UUID projectId) {
+        var restProject = this.projectApplicationService.findById(projectId)
+            .map(project -> new RestProject(project.id(), null, List.of(), null, null, project.name(), List.of()));
+
+        if (restProject.isPresent()) {
+            return new ResponseEntity<>(restProject.get(), HttpStatus.OK);
+        }
+
+        return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+    }
+
+    @PostMapping
+    public ResponseEntity<RestProject> createProject(@RequestParam String name, @RequestParam Optional<String> description) {
+        var createProjectInput = new CreateProjectInput(UUID.randomUUID(), name, List.of());
+        var newProjectPayload = this.projectApplicationService.createProject(createProjectInput);
+
+        if (newProjectPayload instanceof CreateProjectSuccessPayload createProjectSuccessPayload) {
+            var projectDTO = createProjectSuccessPayload.project();
+            var restProject = new RestProject(projectDTO.id(), null, List.of(), null, null, projectDTO.name(), List.of());
+            return new ResponseEntity<>(restProject, HttpStatus.CREATED);
+        }
+        // The specification does not handle other HttpStatus than HttpStatus.CREATED for this endpoint
+        return null;
+    }
+
+    @PutMapping(path = "/{projectId}")
+    public ResponseEntity<RestProject> updateProject(@PathVariable UUID projectId, @RequestParam Optional<String> name, @RequestParam Optional<String> description, @RequestParam Optional<RestBranch> branch) {
```

Comment:
It's a bit odd that the API would expect as an input a RestBranch as a request param. How would that be serialized? Would the end user send the whole object?

I don't care to move forward and merge the PR, it's just odd but it's a first version.

Reply:
Well the specification is definitely not clear on this topic and [the live instance](http://sysml2.intercax.com:9000/docs/#put-/projects/-projectId-) to test things seems to use a different approach with the following body to update a project:

```
{
  "@type": "Project",
  "defaultBranch": {
    "@id": "3fa85f64-5717-4562-b3fc-2c963f66afa6"
  },
  "description": "string",
  "name": "string"
}
```

What part of this is mandatory? what is the schema to use? If I understand things clearly from [their example](https://github.com/Systems-Modeling/SysML-v2-API-Services/blob/master/app/controllers/ProjectController.java#L61) and from this sample it seems that one has to send the full body of the project to update a project. The code of the play based demo seems very odd, I really don't understand the point of something like that:

```
        Project requestedObject = Json.fromJson(requestBodyJson, metamodelProvider.getImplementationClass(Project.class));
        if (requestedObject.getId() != null && !id.equals(requestedObject.getId())) {
            return Results.badRequest();
        }
        requestedObject.setId(id);
```

Given that the identifier if given as a path parameter. And sending back the whole project seems painful to just update a field. By the way, have you seen [this](https://github.com/Systems-Modeling/SysML-v2-API-Services/tree/master/conf/json/schema/metamodel)?

Anyway, let's merge a first version and we will update it later :+1: